google/cloud/internal/curl_impl.cc

```diff
@@ -384,7 +385,14 @@ void CurlImpl::SetUrl(
     url_ = endpoint;
     return;
   }
-  url_ = absl::StrCat(NormalizeEndpoint(endpoint), request.path());
+
+  if (absl::AsciiStrToLower(request.path()).substr(0, endpoint.size()) ==
```

Comment:
Why not check for `https://` and `http://`?  I am not sure this can happen, but maybe we get an upload id pointing to a different endpoint.  For example, if the application used one of the more obscure VPC-SC features where they connection to `foo.private.googleapis.com`.